homeassistant/components/climate/isy994.py

```diff
@@ -0,0 +1,321 @@
+"""
+Support for Insteon Thermostats via ISY994 Platform.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/light.isy994/
+"""
+import logging
+
+from homeassistant.components.climate import (
+    ATTR_CURRENT_HUMIDITY, ATTR_CURRENT_TEMPERATURE, ATTR_FAN_LIST,
+    ATTR_FAN_MODE, ATTR_OPERATION_LIST, ATTR_OPERATION_MODE,
+    ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW, DOMAIN, STATE_AUTO,
+    STATE_COOL, STATE_FAN_ONLY, STATE_HEAT, STATE_IDLE, SUPPORT_FAN_MODE,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE_HIGH,
+    SUPPORT_TARGET_TEMPERATURE_LOW, ClimateDevice)
+from homeassistant.components.isy994 import ISY994_NODES, ISYDevice
+from homeassistant.const import (
+    PRECISION_TENTHS, STATE_OFF, STATE_ON, STATE_UNKNOWN, TEMP_CELSIUS,
+    TEMP_FAHRENHEIT)
+from homeassistant.util.temperature import convert as convert_temperature
+
+_LOGGER = logging.getLogger(__name__)
+
+ISY_CURRENT_TEMPERATURE = 'ST'
+ISY_CURRENT_HUMIDITY = 'CLIHUM'
+ISY_TARGET_TEMP_HIGH = 'CLISPC'
+ISY_TARGET_TEMP_LOW = 'CLISPH'
+ISY_OPERATION_MODE = 'CLIMD'
+ISY_OPERATION_STATE = 'CLIHCS'
+ISY_FAN_MODE = 'CLIFS'
+ISY_UOM = 'UOM'
+
+DEFAULT_MIN_TEMP = 10
+DEFAULT_MAX_TEMP = 30
+
+# Modes not currently supported by 'climate' domain, but may be sent from ISY.
+STATE_PROG_HEAT = 'program_heat'
```

Comment:
You are not allowed to make up states. You need to follow the states of the climate component.

Reply:
Updated to still track the operation mode as reported by the ISY, but to convert these modes into the correct Climate State and hold modes.  Hold modes appear to only be partially implemented at this time (no list of approved hold mode states--e.g. program & temp).